### DISCUSSION TITLE
findIndex Method Issue

### DISCUSSION THREAD

**Discussion by vrbait1107** [babel/babel#11808] [Q&A] [answered]
I am working on Recipe App using Modern Javascript with Babel and Webpack.
I am stuck with findIndex method.

` 

      // This is code of Recipe App
       parseIngredients() {
       const unitLong = ["tablespoons", "tablespoon", "ounces", "ounce","teaspoons","teaspoon", "cups", "pounds" ];

     const unitShort = ["tbsp", "tbsp", "oz","oz","tsp","tsp", "cup", "pound"];

    const newIngredients = this.ingredients.map((item) => {
      let ingredients = item.toLowerCase();

      // Looping unitLong to replace unitLong to unitShort in ingredients.
      unitLong.forEach((unit, i) => {
        ingredients = ingredients.replace(unit, unitShort[i]);
      });

      // Remove Paranthesis
      ingredients = ingredients.replace(/ *\([^)]*\) */g, " ");

      // Parse Ingredients into Count, Unit, Ingredients

      // Spilting Ingredients into Array
      const arrIng = ingredients.split(" ");
      // Checking Wether Unit Present in Ingredient or not

     // Error Here
     const unitIndex = arrIng.findIndex((el2) => unitShort.includes(el2));

      let objIng;

      if (unitIndex > -1) {
        // There is Unit
        const arrCount = arrIng.slice(0, unitIndex);
        let count;
        if (arrCount.length === 1) {
          count = eval(arrIng[0].replace("-", "+"));
        } else {
          count = eval(arrIng.slice(0, unitIndex).join("+"));
        }
        objIng = {
          count: count,
          unit: arrIng(unitIndex),
          ingredients: arrIng.slice(unitIndex + 1).join(),
        };
      } else if (parseInt(arrIng[0], 10)) {
        // If there is no unit but first is number
        objIng = {
          count: parseInt(arrIng[0], 10),
          unit: "",
          ingredients: arrIng.slice(0).join(" "),
        };
      } else if (unitIndex === false) {
        // There is no unit and no count
        objIng = {
          count: 1,
          unit: "",
          ingredients,
        };
      }

      return objIng;
    });

       this.ingredients = newIngredients;
         }
    }
` 

**I think the issue is below code**

`const unitIndex = arrIng.findIndex((el2) => unitShort.includes(el2));`

If I use above  code I get an error 

```
TypeError: arrIng is not a function
    at eval (Recipe.js:126)
    at Array.map (<anonymous>)
    at Recipe.parseIngredients (Recipe.js:92)
    at _callee2$ (index.js:135)
    at tryCatch (runtime.js:65)
    at Generator.invoke [as _invoke] (runtime.js:303)
    at Generator.prototype.<computed> [as next] (runtime.js:117)
    at step (index.js:25) 
    at eval (index.js:25)
```

**Comment by nicolo-ribaudo** [comment, marked as answer]
The problem is this:
```js
    unit: arrIng(unitIndex),
```
You probably meant
```js
    unit: arrIng[unitIndex],
```

(please note that this doesn't have anything to do with Babel)

**Reply by vrbait1107** [reply]
Thank You

**Comment by Patrikoska1** [comment]
`
